mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ImageViewer.java

```diff
@@ -309,17 +318,33 @@ public void extendMenu(JMenuBar menuBar) {
     private synchronized void loadImage(AbstractFile file) throws IOException {
         presenter.getWindowFrame().setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
 
-        Image image;
-        try (ByteArrayOutputStream bout = new ByteArrayOutputStream()) {
-            try (InputStream in = file.getInputStream()) {
-                StreamUtils.copyStream(in, bout);
-                image = imageViewerPanel.getToolkit().createImage(bout.toByteArray());
+        long start = System.currentTimeMillis();
+        BufferedImage image;
+
+        try (ByteArrayOutputStream bout = new ByteArrayOutputStream();
+             InputStream in = file.getInputStream()) {
+
+            StreamUtils.copyStream(in, bout);
+            try (ImageInputStream input = new ByteArrayImageInputStream(bout.toByteArray())) {
+                Iterator<ImageReader> readers = ImageIO.getImageReaders(input);
```

Comment:
can it be that we fail with one reader and succeed with a different one?

Reply:
I will check

Reply:
PR updated